thorlcr/activities/thdiskbaseslave.cpp

```diff
@@ -468,8 +468,8 @@ void CDiskWriteSlaveActivityBase::close()
                 CriticalBlock block(outputCs);
                 // ensure it is released/destroyed after releasing crit, since the IFileIO might involve a final copy and take considerable time.
                 tmpFileIO.setown(outputIO.getClear());
+                mergeStats(closedPartFileStats, tmpFileIO, diskWriteRemoteStatistics);
```

Comment:
Why did these move inside the critical section?  Thread safety?  As long as they don't set any nested stats then it is thread safe, and moving it inside prolongs the length the critsec is held.

I guess it might be to make sure that you don't lose any stats in a race condition when the file is closed, but I think that would generally be ok as long as the final stats are correct.

Reply:
as you say, it could mean that as 1 part closes, and serializeStats enters the crit, it will have lost the stats from the closed part until the next call, meaning numbers go down temporarily.

I wouldn't think the CS held here (only during a close) is consequential timing wise

Reply:
@ghalliday - can move that line outside, but I think it's more correct inside, if cost isn't overly concerning